transport.go

```diff
@@ -38,13 +40,19 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 	span.SetAttributes(semconv.HTTPTargetKey.String(req.URL.Path))
 	span.SetAttributes(semconv.HTTPHostKey.String(req.URL.Host))
 	t.propagator.Inject(traceCtx, propagation.HeaderCarrier(req.Header))
-
+	// maximum size to lumigoSpans RD-7825
+	maxEntrySize := getenv("DEFAULT_MAX_ENTRY_SIZE", 2048)
 	if req.Body != nil {
 		bodyBytes, bodyErr := io.ReadAll(req.Body)
 		if bodyErr != nil {
 			logger.WithError(bodyErr).Error("failed to parse request body")
 		}
-		span.SetAttributes(attribute.String("http.request_body", string(bodyBytes)))
+
+		if len(bodyBytes) > maxEntrySize {
+			span.SetAttributes(attribute.String("http.request_body", string(bodyBytes[:maxEntrySize])))
```

Comment:
I think it still means that we save the whole `bodyBytes` in the memory.  If the response's body is huge then we might use too much memory, can you read less data instead of reading all and then slicing it?
In python it will be something like: `f.read(2048)` instead of `f.read()[:2048]`

Is this what you meant with the comment: "// TODO: add limit reader RD-7825"?